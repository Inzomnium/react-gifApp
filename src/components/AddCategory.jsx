import { useState } from "react";





export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setinputValue] = useState('') 

  const enInputChange = (event) => {

    setinputValue(event.target.value);

  }

    const onSubmit = (event) => {
        event.preventDefault();
        //previene el gatillo de la funcion cuando hay 0 o 1 caracter en el inmput
        if (inputValue.trim(). length <= 1) return;
        //setCategories( categories  => [inputValue, ...categories])
        onNewCategory( inputValue.trim() );
        setinputValue(' ')
        
    }

    return (
        <form  onSubmit={onSubmit}>

        <input 
         type="text"
         placeholder="ingresa texto"
         value={inputValue}
         onChange={ (event) => enInputChange(event) }
        />
        

        </form>
    )


}