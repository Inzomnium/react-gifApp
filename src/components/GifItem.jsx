

export const GifItem = ({title, url, id}) => {

    

  return (

   
    <div
    id={id}
    className="card">
       <img src={url} alt={title} />
       <p>{title}</p>
    </div>
   
    
  )
}
