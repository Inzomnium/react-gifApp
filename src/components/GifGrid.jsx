
import { useState, useEffect } from "react";

import { GifItem } from "./GifItem";

//import { getGifs } from "../helpers/getGifs"
import { useFetchGifs } from "../hooks/useFetchGifs";



export const GifGrid =  ({ category}) => {

        //custom hook
    const {images, isLoading} = useFetchGifs(category);

    
   
    
    return (
    <>
        <h3>{ category }</h3>
        
        {
            isLoading && (          
                <img width="300" src="/src/images/loading.gif" />
            ) 
        }

        
        <div className="card-grid">
            { 
            images.map((image) => (
                
               <GifItem 
               key={image.id}
               {... image}
                />

            ))
            }
            
            
           

        </div>
        
    
    
    </>
  )
}
