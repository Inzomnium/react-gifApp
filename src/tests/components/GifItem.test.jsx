import { render, screen } from "@testing-library/react";
import { string } from "prop-types";
import { GifItem } from "../../components";

describe('Pruebas en GifItem', () => { 
    const title = 'title test string';
    const url = 'https://www.testurl/testurl';
    const {container} = render( <GifItem title={title} url={url} /> );

    test('should match with the snapshot', () => {

       
       expect(container).toMatchSnapshot()


     });

     test('should title be a string', () => { 

        expect(typeof title).toBe('string')


      })
      test('should URL be contained in the snapshot', () => { 

        render( <GifItem title={title} url={url} /> );

        expect(screen.getByRole('img').src).toBe(url)
        

        

      })
      test('should img alt title equal to title', () => { 

        render( <GifItem title={title} url={url} /> );

        expect(screen.getByRole('img').alt).toBe(title)

       })


 });