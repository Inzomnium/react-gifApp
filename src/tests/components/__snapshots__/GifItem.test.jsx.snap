// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title test string"
      src="https://www.testurl/testurl"
    />
    <p>
      title test string
    </p>
  </div>
</div>
`;
